### PR TITLE
[MRG] Removed Py2 related compatibility code

### PR DESCRIPTION
**Pull request by kchawla-pi**
Since we dont support Python2 aymore, we can start removing the compatibility shims

### CODE REVIEW

**Comment by kchawla-pi** [comment]
CircleCI is passing: https://circleci.com/gh/kchawla-pi/nilearn/1063

**Comment by GaelVaroquaux** [comment]
Thanks Kshitij!

Do you know why we are no longer seeing circle in the status?

File: nilearn/_utils/compat.py [hunk 1]
```diff
@@ -1,76 +1,10 @@
...
-
-
-import joblib
-
-Memory = joblib.Memory
-Parallel = joblib.Parallel
-hash = joblib.hash
-delayed = joblib.delayed
-cpu_count = joblib.cpu_count
+def md5_hash(string):
```
Comment (GaelVaroquaux) [inline nilearn/_utils/compat.py]: I think that we should move this helper function in the only file where it is used: it is no longer a compatibility layer.

**Comment by GaelVaroquaux** [comment]
We got rid of a file! :). That's great.

Merging!

**Pull request merged**